src/renderers/shared/fiber/ReactChildFiber.js

```diff
@@ -43,6 +43,38 @@ if (__DEV__) {
   var getComponentName = require('getComponentName');
   var warning = require('fbjs/lib/warning');
   var didWarnAboutMaps = false;
+  /**
+   * Warn if there's no key explicitly set on dynamic arrays of children or
+   * object keys are not valid. This allows us to keep track of children between
+   * updates.
+   */
+  var ownerHasKeyUseWarning = {};
+
+  var warnForMissingKey = (child: ReactElement) => {
+    if (!child._store || child._store.validated || child.key != null) {
+      return;
+    }
+    child._store.validated = true;
+
+    var memoizer = ownerHasKeyUseWarning.uniqueKey ||
+      (ownerHasKeyUseWarning.uniqueKey = {});
```

Comment:
Do you know why this is a nested object and constructed dynamically? I don't understand the intention of the original source. Maybe worth simplifying.